### ISSUE TITLE
Generated classes should respect properties upper/lower case formatting

### ISSUE THREAD

**Issue by plm75** [apollographql/apollo-ios#72]
Server side schema: 
```
type MyModel {
  id: String 
  imageURL: String
}
```

iOS - generated class:
```swift 
public final class MyQuery: GraphQLQuery {
   [ ....... ]
  public struct MyModel: GraphQLMappable {
    public let id: String?
    public let imageUrl: String?     // <----------- imageUrl != imageURL
  }
}
```

Could it be possible in future versions to try and have the exact same property names in the generated classes and in the schema ? Thanks

**Comment by martijnwalraven** [comment]
Yes, we have to find a better way to deal with this. Not converting names at all means models may end up containing field names with underscores or other conventions if that is what the schema uses.

There is something to be said for just passing those names through, but using underscored properties in Swift just feels wrong :)

For that reason, I think I prefer making the conversion smarter to keep the original casing in these situations. What do you think?

**Comment by plm75** [comment]
@martijnwalraven : TBH, IMHO, you shouldn't worry/burden yourself with this: just pass everything as is and let the developer be responsible for making sure his/her naming scheme works.  Guideline/best practices plus warnings about edges cases in documentation should be fine.  I'd rather you guys focus on improving core functionalities instead of second-guessing/cleaning after developers 😄

**Comment by martijnwalraven** [comment]
Yeah, you're probably right. The schema isn't always under the mobile developer's control however, so I wanted to make it convenient to still get a nice API by default.

**Comment by highthem** [comment]
I have the same issue!!
maybe you need to add an option for generated code to respect or not the way pramas are written in the schema.json

**Comment by calvincestari** [comment]
Hi 👋🏻 - we've taken this into consideration and decided to defer it to a future 1.x release. We're not rejecting nor closing this issue but it's not on the roadmap yet. If this issue is important to you please let us know _in this issue_ and it will help with prioritization - thank you!
